lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -356,37 +356,51 @@ export class UserAgentApplication {
       scopes = this.filterScopes(scopes);
     }
 
-    this._loginInProgress = true;
-    
-    this.authorityInstance.ResolveEndpointsAsync()
-      .then(() => {
-        const authenticationRequest = new AuthenticationRequestParameters(this.authorityInstance, this.clientId, scopes, ResponseTypes.id_token, this._redirectUri, this._state);
-        if (extraQueryParameters) {
-          authenticationRequest.extraQueryParameters = extraQueryParameters;
-        }
-
-        var loginStartPage = this._cacheStorage.getItem(Constants.angularLoginRequest);
-        if (!loginStartPage || loginStartPage === "") {
-            loginStartPage = window.location.href;
-        }
-        else {
-            this._cacheStorage.setItem(Constants.angularLoginRequest, "")
-        }
-
-        this._cacheStorage.setItem(Constants.loginRequest, loginStartPage, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.loginError, "");
-        this._cacheStorage.setItem(Constants.stateLogin, authenticationRequest.state, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.nonceIdToken, authenticationRequest.nonce, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.msalError, "");
-        this._cacheStorage.setItem(Constants.msalErrorDescription, "");
-        const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
-        if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
-          this._cacheStorage.setItem(authorityKey, this.authority);
-        }
-
-        const urlNavigate = authenticationRequest.createNavigateUrl(scopes)  + Constants.prompt_select_account + Constants.response_mode_fragment;
-        this.promptUser(urlNavigate);
-      });
+      this._loginInProgress = true;
+      const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
+      var idTokenObject;
+      if(adalIdToken) {
+           idTokenObject = Utils.extractIdToken(adalIdToken);
+      }
+      if (idTokenObject && idTokenObject.upn ) {
+          console.log("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
+          var extraQueryParam= "&" + Constants.login_hint+ "=" + idTokenObject.upn + "&" + Constants.domain_hint + "=" + Constants.organizations;
+          this.acquireTokenSilent([this.clientId], this.authority, this.getUser(), extraQueryParam)
+              .then((idToken) => {
+                  this._tokenReceivedCallback.call(this, "", idToken, "", "idToken", "");
+              });
```

Comment:
Handle the error condition

Reply:
In the tokenReceivedCallback, the tokenType should be Constants.idToken which resolves to id_token not idToken.

Reply:
Also, In the success case, the errorDesc and error fields should be null and the in the success case, the token field should be null. This will help the user in writing code such as if(token) or if(error && errorDesc). Sending error and errorDesc as an empty string in case of a successful response will be misleading.

Reply:
I agree with Aashima, handling the error condition is most important. Otherwise the user will never be prompted for login.

Reply:
@rohitnarula7176
1.  I will change the code to use the constant for idToken.
2. There was a bug in ATS that I fixed as part for this PR. The fix is -> If user is passing only client_id and no user object then it will just get the idtoken else "idtoken token". So it will never be called for access_token. This needs to be fixed in interactive acquireToken apis as well but we are planning to do it post ignite.
3. I am change the code to pass null for error and errorDesc in error case instead of null.

Reply:
I believe MSAL did not support SSO by using login_hint and domain_hint to make use of an existing session previously. So it was not a bug but a not supported scenario previously.